splash/network_manager.py

```diff
@@ -363,9 +367,23 @@ def __init__(self, filters_path, allowed_schemes, verbosity):
                 AdblockMiddleware(self.adblock_rules, verbosity=verbosity)
             )
 
+        self.response_middlewares.append(ContentTypeMiddleware(self.verbosity))
+
+    def run_response_middlewares(self):
+        reply = self.sender()
+        render_options = self._getRenderOptions(reply.request())
+        if render_options:
+            for middleware in self.response_middlewares:
+                middleware.process(reply, render_options)
+        reply.metaDataChanged.disconnect(self.run_response_middlewares)
```

Comment:
It could be better to disconnect the callback in a 'finally' block. 
By the way, why is disconnecting needed?